atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresGetRange.java

```diff
@@ -138,7 +139,8 @@ public PostgresGetRange(PostgresPrefixedTableNames prefixedTableNames,
                                                long timestamp) {
         int maxRowsPerPage = RangeHelpers.getMaxRowsPerPage(rangeRequest);
         int cellsPerRowEstimate = getCellsPerRowEstimate(tableRef, rangeRequest);
-        int maxCellsPerPage = maxRowsPerPage * cellsPerRowEstimate + 1;
+        int maxCellsPerPage = Math.min(
```

Comment:
Take a look at BatchSizeIncreasingIterator.

I don't think we need logging here.